### ISSUE TITLE
Request priority not working with ensemble model containing python backend and ONNX model 

### ISSUE THREAD

**Issue by chandanidoshi** [triton-inference-server/server#6055]
**Description**

We are using an ensemble model with the first two steps being a python backend tokenizer and an ONNX model to get embeddings and have enabled dynamic batching in the ONNX model. We want to set request priority in the ONNX model but it seems like the priority level is not getting passed through to it in the ensemble. We have checked that priority works when we make requests directly to the ONNX model and it also seems to be working if the ONNX model is the first step in the ensemble scheduler. What should we do to ensure the request priority gets passed through to the ONNX model when we include the tokenizer before the ONNX model in the ensemble scheduler?

**Triton Information**
What version of Triton are you using?

23.05

Are you using the Triton container or did you build it yourself?

Yes, we're using the Triton container 

**To Reproduce**

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

ONNX config: 

```
name: "all-mpnet-base-v2-onnx"
max_batch_size: 256
input {
  name: "input_ids"
  data_type: TYPE_INT64
  dims: -1
}
input {
  name: "attention_mask"
  data_type: TYPE_INT64
  dims: -1
}
output {
  name: "last_hidden_state"
  data_type: TYPE_FP32
  dims: -1
  dims: 768
}
instance_group {
  count: 1
  kind: KIND_GPU
}
dynamic_batching {
  max_queue_delay_microseconds: 10000
  priority_levels: 3
  default_priority_level: 3
}
backend: "onnxruntime"
```

Ensemble config:

```
name: "all-mpnet-base-v2-inference"
platform: "ensemble"
max_batch_size: 256
input {
  name: "input"
  data_type: TYPE_STRING
  dims: 1
}
output {
  name: "normalized_embeddings"
  data_type: TYPE_FP32
  dims: 768
}
ensemble_scheduling {
  step {
    model_name: "all-mpnet-base-v2-tokenizer"
    model_version: -1
    input_map {
      key: "TEXT"
      value: "input"
    }
    output_map {
      key: "attention_mask"
      value: "attention_mask"
    }
    output_map {
      key: "input_ids"
      value: "input_ids"
    }
  }
  step {
    model_name: "all-mpnet-base-v2-onnx"
    model_version: -1
    input_map {
      key: "attention_mask"
      value: "attention_mask"
    }
    input_map {
      key: "input_ids"
      value: "input_ids"
    }
    output_map {
      key: "last_hidden_state"
      value: "last_hidden_state"
    }
  }
  step {
    model_name: "mean-pooling"
    model_version: -1
    input_map {
      key: "attention_mask"
      value: "attention_mask"
    }
    input_map {
      key: "last_hidden_state"
      value: "last_hidden_state"
    }
    output_map {
      key: "embeddings"
      value: "embeddings"
    }
  }
}
```

**Comment by Tabrizian** [comment]
As far as I can tell from the code, it looks like priority is indeed being passed to the first stage: https://github.com/triton-inference-server/core/blob/449107ef4387c1da48d8c1a2c7b1a664b69d8362/src/ensemble_scheduler.cc#L952

Maybe the issue is that it is not being passed further down the pipeline. @GuanLuo do you think it makes sense to pass priority to other models (not just first model in ensemble) or there is a reason for not doing so?

**Comment by GuanLuo** [comment]
I think that is the common function called on every step in the ensemble, so the same priority of the ensemble request will be passed to all inferences within the pipeline.

**Comment by chandanidoshi** [comment]
We tried logging the request parameters in the python backend steps using:
```
logger.log_info(f"Request parameters: {request.parameters()}")
```
The logs show that this is an empty dictionary.
```
I0718 19:45:30.224659 1 model.py:76] Request parameters: {}
```

**Comment by Tabrizian** [comment]
The `request.parameters` would not include the parameters that are consumed by the core (such as sequence id, priority, binary data). Currently, there is no way to access this parameter. Looks like the current expectation is to pass the priority to all the models. Are you able to provide a model repository? I'll file a bug to look into this issue.

**Comment by chandanidoshi** [comment]
You can download the model repository from here: [triton_models](https://assets.heyday.xyz/triton_model_repository/models.zip)

Our main goal is to ensure that the priority level gets passed to our ONNX model when it is a part of an ensemble model.
